### ISSUE TITLE
Add integration tests for syntax, analysis and compiler

### ISSUE THREAD

**Issue by MeAmAnUsername** [MeAmAnUsername/pie#233]
**Summary**
Add integration tests for syntax, analysis and compiler

**Todo**
- [ ] Create/find full PIE programs to use as integration tests
- [ ] Add integration tests for syntax
- [ ] Add integration tests for static semantics
- [ ] Add integration tests for compiler

**Reason**
Right now there are only unit tests, add some integration tests to make sure everything works together

**Implementation**
Basically just write a (few) full program(s) and make an SPT test that tests the syntax, an SPT test that tests the static semantics, and an e2e test in the compiler
